### DISCUSSION TITLE
tilestorage

### DISCUSSION THREAD

**Discussion by paprophet** [iTwin/itwinjs-core#5805] [Q&A] [answered]
now i change oss storage area.(https://github.com/iTwin/itwinjs-core/discussions/5714)
but i open the model , and  show some  tiles can not download in frontend
use the minio storage server ,the problem exist too .
 but i double click the link ,and the tile file can download in the web brower.
![storagereaderror](https://github.com/iTwin/itwinjs-core/assets/18022720/c20ab1d7-9e60-415b-8590-dc3479ef6dfa)

**Comment by aruniverse** [comment]
Tiles are dynamically generated. We will attempt to fetch them from the tile storage cache if they exist; if they don't they will be dynamically generated. Missing tiles here should not be an issue

cc @austeja-bentley @robertas-kerpys @paulius-valiunas

**Reply by paprophet** [reply]
but the model can not show complelte in web.
the red line area can not request the tile.
![tt2](https://github.com/iTwin/itwinjs-core/assets/18022720/8ea3f0c1-8d10-4e73-938a-bf337b8afcc7)

**Reply by paulius-valiunas** [reply, marked as answer]
your first screenshot indicates that most tiles were downloaded successfully, so I believe the tile loading mechanism is not broken. Can you check if it works better today? My guess is that there was some temporary downtime on the OSS bucket while you were trying to fetch those tiles.

Also, you say your browser downloaded something when you double clicked the URL. That is not normally possible because OSS requires setting the authorization header to download an object. I see two possibilities here:

1. your bucket is misconfigured and allows accessing every object from anywhere without any authorization (I'm not sure if that's possible)
2. your browser just downloaded the 404 response body for you since it couldn't display it natively

When you double-click the link, please open the browser debugger's network tab, refresh the page and observe the response code that it returns. Most browsers don't make it obvious whether the URL entered in the address bar returned a successful response or an error.
